### PR TITLE
[CORE 3978]: Channel Token Authorization

### PR DESCRIPTION
**Pull request by gupta-himanshu**
JIRA Ticket: https://flowio.atlassian.net/browse/CORE-3978

### CODE REVIEW

**Review by mikeclimbrock — APPROVED** [review]
overall, lgtm!

**Comment by mikeclimbrock** [comment]
@gupta-himanshu - let's aim to deploy tomorrow, 12/28.

File: app/auth/ChannelAuth.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    authFuture.map { _ =>
+      Some(
+        token.copy(
+          channelId = token.channelId,
+          environment = token.environment
+        )
+      )
+    }.recover {
```
Comment (mbryzek) [inline app/auth/ChannelAuth.scala:61]: We have been using recoverWith here which then lets you replace the
```
sys.error
```
below with
```
Future.failed(ex)
```

**Review by mbryzek — APPROVED** [review]

File: app/auth/ChannelAuth.scala (hunk 1 above)
Comment (gupta-himanshu) [reply app/auth/ChannelAuth.scala:61]: ack

**Comment by gupta-himanshu** [comment]
> @gupta-himanshu - let's aim to deploy tomorrow, 12/28.

ack

**Pull request merged**